### PR TITLE
Add boundary condition tests for DateInterval.contains

### PR DESCRIPTION
**Pull request by ArihantMarwaha**
### Description
This PR adds boundary condition tests for `DateInterval.contains(_:)` to verify that the start and end dates are correctly evaluated as being inside the interval, and that dates just outside the range (by 1 millisecond) are correctly excluded.

### Checklist
- [x] Run unit tests locally (`FoundationEssentialsTests` passed)
- [x] Follow contribution and code style guidelines

### CODE REVIEW

File: Tests/FoundationEssentialsTests/DateIntervalTests.swift [hunk 1]
```diff
@@ -155,6 +155,12 @@ private struct DateIntervalTests {
         // dateWithString("2009-05-17 14:49:47 -0700")
         let earlierStart = Date(timeIntervalSinceReferenceDate: 264289787.0)
         #expect(!testInterval.contains(earlierStart))
+        
+        //Boundary cases
+        #expect(testInterval.contains(start))
+        #expect(testInterval.contains(testInterval.end))
+        #expect(!testInterval.contains(start - 0.001))
```
Comment (parkera) [inline Tests/FoundationEssentialsTests/DateIntervalTests.swift]: I think this is effectively the same test as `earlierStart` above.

File: Tests/FoundationEssentialsTests/DateIntervalTests.swift [hunk 2]
```diff
@@ -155,6 +155,12 @@ private struct DateIntervalTests {
...
+        #expect(!testInterval.contains(testInterval.end + 0.001))
```
Comment (parkera) [inline Tests/FoundationEssentialsTests/DateIntervalTests.swift]: Maybe following the style here, we use the `timeIntervalSinceReferenceDate` to create a date past the end for testing.

File: Tests/FoundationEssentialsTests/DateIntervalTests.swift (hunk 1 above)
Comment (ArihantMarwaha) [reply Tests/FoundationEssentialsTests/DateIntervalTests.swift]: removed the redundant `start - 0.001 `check since it is already covered by the `earlierStart` test

File: Tests/FoundationEssentialsTests/DateIntervalTests.swift (hunk 2 above)
Comment (ArihantMarwaha) [reply Tests/FoundationEssentialsTests/DateIntervalTests.swift]: @parkera
Refactored this check to construct the past-end date using the timeIntervalSinceReferenceDate style to match the rest of the file.

**Comment by parkera** [comment]
@swift-ci test

**Review by parkera — APPROVED** [review]

**Comment by ArihantMarwaha** [comment]
@parkera, just wanted to know if this PR requires any additions to be merged or run any additional tests?

**Comment by parkera** [comment]
Let me see if I can kick the CI again here

**Comment by ArihantMarwaha** [comment]
When do these additions get merged into production @parkera ?
